From 736f46182026753c5f8cb75b0b599aa592b8e1cc Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 22 Mar 2017 09:17:30 +0200
Subject: [PATCH 08/19] arm: mx6: cl_som_mx6: mmc rescan fix and etc.

1) Fix mmc rescan issue.
2) Release unused gpios.
3) Fix a typo mistake.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 board/compulab/cm_fx6/cm_fx6.c | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/board/compulab/cm_fx6/cm_fx6.c b/board/compulab/cm_fx6/cm_fx6.c
index bcd52c3..e4e38b9 100644
--- a/board/compulab/cm_fx6/cm_fx6.c
+++ b/board/compulab/cm_fx6/cm_fx6.c
@@ -541,6 +541,13 @@ int board_eth_init(bd_t *bis)
 	udelay(1000);
 	SETUP_IOMUX_PADS(enet_pads_up);
 
+	gpio_free(IMX_GPIO_NR(6, 30));
+	gpio_free(IMX_GPIO_NR(6, 25));
+	gpio_free(IMX_GPIO_NR(6, 27));
+	gpio_free(IMX_GPIO_NR(6, 28));
+	gpio_free(IMX_GPIO_NR(6, 29));
+	gpio_free(IMX_GPIO_NR(6, 24));
+
 	enable_enet_clk(1);
 	return cpu_eth_init(bis);
 }
@@ -622,10 +629,27 @@ static enum mxc_clock usdhc_clk[CONFIG_SYS_FSL_USDHC_NUM] = {
 	MXC_ESDHC4_CLK,
 };
 
+static iomux_v3_cfg_t const usdhc_wifi_pads[] = {
+	IOMUX_PADS(PAD_NANDF_CS3__GPIO6_IO16	| MUX_PAD_CTRL(NO_PAD_CTRL)),
+	IOMUX_PADS(PAD_EIM_DA1__GPIO3_IO01	| MUX_PAD_CTRL(NO_PAD_CTRL)),
+};
+
+static void usdhc_wifi_mmc_init(void) {
+
+	SETUP_IOMUX_PADS(usdhc_wifi_pads);
+
+	gpio_request(IMX_GPIO_NR(6, 16),"6_16");
+	gpio_request(IMX_GPIO_NR(3, 1),"3_01");
+
+	gpio_direction_output(IMX_GPIO_NR(6, 16), 1);
+	gpio_direction_output(IMX_GPIO_NR(3, 1), 1);
+}
+
 int board_mmc_init(bd_t *bis)
 {
 	int i;
 
+	usdhc_wifi_mmc_init();
 	cm_fx6_set_usdhc_iomux();
 	for (i = 0; i < (CONFIG_SYS_FSL_USDHC_NUM - nand_enabled) ; i++) {
 		usdhc_cfg[i].sdhc_clk = mxc_get_clock(usdhc_clk[i]);
@@ -705,7 +729,7 @@ int ft_board_setup(void *blob, bd_t *bd)
 	} else {
 		/* emmc enable */
 		fdt_status_okay_by_alias(blob, usdhc4_path);
-		/* Disable an rtc that makes use of a nand pin for irq */
+		/* Enable an rtc that makes use of a nand pin for irq */
 		fdt_status_okay_by_alias(blob, rtc_nand);
 	}
 
-- 
1.9.1

